source/daplink/settings/settings_rom.c

```diff
@@ -61,10 +61,10 @@ COMPILER_ASSERT(sizeof(cfg_setting_t) < SECTOR_BUFFER_SIZE);
 COMPILER_ASSERT(SECTOR_BUFFER_SIZE % 8 == 0);
 
 // Configuration ROM
-#if defined(__CC_ARM) || defined(__ARMCC_VERSION)
+#if defined(__CC_ARM)
 static volatile cfg_setting_t config_rom __attribute__((section("cfgrom"),zero_init));
 #else
-static volatile cfg_setting_t config_rom __attribute__((section("cfgrom")));
+static volatile cfg_setting_t config_rom __attribute__((section(".bss.cfgrom")));
```

Comment:
This will break the persistent settings. `cfgrom` must reside in internal flash. For gcc, there is a `m_cfgrom` memory defined, and a `.cfgrom` output section that places the `cfgrom` input section into `m_cfgrom`.

Reply:
I pushed a version without that change